### DISCUSSION TITLE
Update model with ground truth?

### DISCUSSION THREAD

**Discussion by quannabe** [winedarksea/AutoTS#117] [Q&A]
For example, I have created a new model with monthly data through March and then April becomes available. Is it possible to update the model without retraining to incorporate this new, known value?

**Comment by winedarksea** [comment]
The simplest way is to run the `model_forecast` function having extracted the best_model params from the AutoTS object. Pass them in and run with your updated DF. There's a partial example on that [here](https://github.com/winedarksea/AutoTS/blob/master/extended_tutorial.md#running-just-one-model).

You can also check out the [production_example.py](https://github.com/winedarksea/AutoTS/blob/master/production_example.py). The key thing to adjust there is `evolve=False`. Basically the way I use it is to load a template, run a (smaller) number of generations, and then choose the best model again. You can modify so that loads a template and chooses from only a small subset of models (give it a chance to switch models, but only among those you have allowed), and of course finally to only run and predict on one model. The number of options can make it a bit confusing. Let me know if that helps @quannabe !

**Reply by quannabe** [reply]
Got it, both seem like good options!

In your first option, I assume I would then be updating `df_train`, correct?

**Reply by winedarksea** [reply]
Yes. Although good point that name is a bit confusing. `df_train` is yes, just the dataframe of all the historical data you have available when you run it.

Just to confuse you more, some models will 'retrain' when run with model_forecast (basically GluonTS and anything ending in 'Regression'), but unless your data has shifted drastically, it shouldn't significantly change the behavior. If you are not satisfied with that behavior, then there's always diving into the model objects directly for each type of model, although the documentation is much weaker at that level.

**Reply by quannabe** [reply]
Cool, thank you for confirming and the detailed response!

**Reply by quannabe** [reply]
Just following up on this one. 

I see the error: `future_regressor row count does not match length of training data` when trying this as I described above. 

Since the resulting model gives me `model.regressor_used=False` I assume this means the regressor I added is ignored? Does this mean that my regressor (weather forecast) is being ignored because the length is longer?

**Reply by quannabe** [reply]
Hmm, maybe I am mistaken, but it does seem like the regressor is actually being used as my model results differ if I omit the `future_regressor` from model.fit()

**Reply by winedarksea** [reply]
So model.regressor_used is... I have no idea when that is from. It isn't ever filled. What is used is model.used_regressor_check but it is currently semi-broken (may not always have the correct answer). As they are no longer needed for generating anything except as an FYI to users, I haven't bothered to fix them.
But I did just bother to fix them so it should be working in 0.4.1 released in a week or so.

Until such time, any ModelParameters with `regression_type=='User'` are those using a regressor. See the information on Simulation mode as well.

As for `future_regressor row count does not match length of training data` that is just a warning. What it will then do is attempt to reindex your data (possibly dropping extra data if your regressor has more dates than your other df) or filling with NaN any missing parts of the regressor. You should check and see if they cover the exact same time period, or if one is shorter or longer.
